pkg/reconciler/dependencybuild/buildrecipeyaml.go

```diff
@@ -19,7 +19,14 @@ spec:
     value: quay.io/sdouglas/hacbs-jdk11-builder:latest
   - name: GOALS
     value:
-    - build
+      - -DskipTests
+      - clean
+      - install
+      - -Denforcer.skip
+  - name: JAVA_VERSION
+    value: ""
+  - name: TOOL_VERSION
+    value: ""
```

Comment:
@stuartwdouglas Should I be adding default empty values for Maven. The file `deploy/base/maven-v0.2.yaml` currently doesn't have that added.

If I understand correctly, `TOOL_VERSION` is going away, but `JAVA_VERSION` may be useful for the Maven case, also?

Reply:
aside from adding TOOL_VERSION and JAVA_VERSION, you added entries to the GOALS param array

technically speaking changing an existing param on the maven recipe would not be part of adding gradle

but I'll defer to @stuartwdouglas - are you good with that aspect of this change as well ?

Reply:
@gabemontero I believe it was your original commit when `pkg/reconciler/dependencybuild/buildrecipeyaml.go` was created that had Maven `GOALS` set to `- build` which can't be right since that's a Gradle task. Correct me if I'm wrong.

I tried updating it in my pull request to match the existing maven goals. I don't know if my change is correct, but the original upstream was not correct either.

I think a couple of these maven changes probably belong in their own pull request anyway.

Reply:
ah you are right @dwalluck now that I look at main brainch

but it shows up as a diff in the PR view ... again, you branch is a bit behind main and that is confusing me.  4 commits behind based on what https://github.com/dwalluck/jvm-build-service/tree/PLNSRVCE-346 tells me.

yeah per my recent comments elsewhere, I think some commit squashing, and another rebase, are in order.

Reply:
I am fine with these for now in terms of just getting this feature in, but I do want to revisit this in the future. I need to capture all this in a JIRA but basically what I am thinking is:
- General purpose builder images with both maven and gradle
- Programatically assemble these tasks keeping them as close as possible (so we can re-use code between both)
- System wide config as to which images have which versions, and we use this to map builder images to requests
- The actual tool and command line to be selected programatically and set as parameters, rather than something hard coded into the image.

We are not really in a position to start on this yet though, we need this in and my config PR in and then we can begin to look at how this will all fit together.